### PR TITLE
fix: merge lock/fallback flow init

### PR DESCRIPTION
**Pull request by acezard**

### CODE REVIEW

File: src/screens/home/components/HomeView.js [hunk 1]
```diff
@@ -185,49 +184,41 @@ const HomeView = ({ route, navigation, setLauncherContext, setBarStyle }) => {
...
+        ? 'flat'
+        : 'nested'
 
+      let navigationObject = null
+
+      if (uri) {
+        const cozyAppFallbackURL =
+          'http://drive.dev.192-168-1-65.nip.io:8080/#/folder' // cozyAppFallbackURLInitialParam.consume()
         if (cozyAppFallbackURL) {
-          setShouldWaitCozyApp(true)
```
Comment (Ldoppea) [inline src/screens/home/components/HomeView.js:194]: Are you sure you want to remove that line? this is the one that allows the HomeView to wait until CozyAppScreen is loaded

File: src/screens/home/components/HomeView.js [hunk 2]
```diff
@@ -185,49 +184,41 @@ const HomeView = ({ route, navigation, setLauncherContext, setBarStyle }) => {
...
+    async function handleSecurityFlowAndCozyAppFallback() {
+      const subdomainType = client.capabilities?.flat_subdomains
+        ? 'flat'
+        : 'nested'
 
+      let navigationObject = null
+
+      if (uri) {
+        const cozyAppFallbackURL =
+          'http://drive.dev.192-168-1-65.nip.io:8080/#/folder' // cozyAppFallbackURLInitialParam.consume()
```
Comment (Ldoppea) [inline src/screens/home/components/HomeView.js:197]: Don't forget to revert that line when dev is done

File: src/app/view/Secure/PinPrompt.tsx [hunk 3]
```diff
@@ -1,18 +1,33 @@
...
   const handleSetPinCode = (): void => {
-    navigate(routes.setPin)
+    navigate(routes.setPin, { onSuccess: props.route.params?.onSuccess })
   }
 
   const handleIgnorePinCode = (): void => {
-    navigate(routes.home)
+    try {
+      if (!props.route.params?.onSuccess)
+        throw new Error('No onSuccess callback given to PinPrompt')
```
Comment (Ldoppea) [inline src/app/view/Secure/PinPrompt.tsx:25]: the `if` condition is different from the one on `PasswordPrompt`, is that wanted?

File: src/app/view/Secure/PasswordPrompt.tsx [hunk 4]
```diff
@@ -1,13 +1,29 @@
...
   const handleSetPassword = (): void => {
-    navigate(routes.setPassword)
+    if (
+      props.route.params &&
+      typeof props.route.params.onSuccess === 'function'
+    ) {
+      navigate(routes.setPin, { onSuccess: props.route.params.onSuccess })
+    } else {
+      // Handle the case where params or onSuccess is undefined
+      // This should not happen and is considered a critical error
```
[the remark this answers is not in the document]
Comment (acezard) [reply src/app/view/Secure/PasswordPrompt.tsx:24]: I should consolidate the two functions to be the same (import it)

File: src/app/view/Secure/PinPrompt.tsx (hunk 3 above)
Comment (acezard) [reply src/app/view/Secure/PinPrompt.tsx:25]: I should consolidate the two functions to be the same (import it)

File: src/app/view/Secure/PasswordPrompt.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (acezard) [reply src/app/view/Secure/PasswordPrompt.tsx]: updated

File: src/app/view/Secure/PinPrompt.tsx (hunk 3 above)
Comment (acezard) [reply src/app/view/Secure/PinPrompt.tsx:25]: updated

**Review by Ldoppea — APPROVED** [review]

**Pull request merged**